ProjFS.Mac/PrjFSKext/VirtualizationRoots.cpp

```diff
@@ -510,17 +518,16 @@ VirtualizationRootResult VirtualizationRoot_RegisterProviderForPath(PrjFSProvide
         }
     }
     
-    if (NULLVP != virtualizationRootVNode)
+    if (VirtualizationRoot_IsValidRootHandle(rootIndex))
     {
-        vnode_put(virtualizationRootVNode);
+        vfs_setauthcache_ttl(vnode_mount(virtualizationRootVNode), 0);
```

Comment:
Ideally, I'd like to verify this in tests by requiring `vnode_mount()` to only be called on vnodes with an active refcount, but this is currently not practical as a vast number of unit tests benignly violate this elsewhere.